### DISCUSSION TITLE
Is there a way to execute a script at startup?

### DISCUSSION THREAD

**Discussion by sommerper** [niri-wm/niri#266] [Q&A] [answered]
Hi!

I recently discovered this wonderful compositor and I'm trying to figure out if it is at a usable state for me. So I'll likely ask a bunch of questions over the next days if you don't mind.

Comparing to hyprland where it is possible to run a script from the config file at start-up like so:
```
exec-once=bash ~/.config/hypr/start.sh
```
Is there an equivalent method in Niri?
I know it possible to set up services as you describe here https://github.com/YaLTeR/niri/wiki/Example-systemd-Setup but I think it's a little more convenient if I could just execute a script :-)

And thank you for this project!

**Comment by YaLTeR** [comment, marked as answer]
Hello, yeah, it's possible:

```
spawn-at-startup "bash" "myscript.sh"
```

at the top level. Note that the ~ in the second argument will not be expanded currently, you need to write out the home dir.

**Reply by sommerper** [reply]
Thanks for the quick reply!
I completely missed that one. I see it now in the default config file.
Sorry about that!

**Comment by dead0user** [comment]
`spawn-at-startup` don't works correct on my setup:

```
spawn-at-startup "waybar" "-c" "~/.config/waybar/config_niri"
spawn-at-startup "swaybg" "-i" "~/Pictures/woman-7795469_1920.jpg"
```
but if I run waybar or swaybg manually in a background, it works correct

**Reply by YaLTeR** [reply]
`~` is not expanded in arguments, you need to write it out. See https://github.com/YaLTeR/niri/wiki/Configuration:-Key-Bindings#spawn for examples

**Reply by dinotheextinct** [reply]
> `~` is not expanded in arguments, you need to write it out. See https://github.com/YaLTeR/niri/wiki/Configuration:-Key-Bindings#spawn for examples

Just why? I have never seen something like this. Is it supposed to be quotation game for some bored days??? This is insane.

**Reply by YaLTeR** [reply]
https://niri-wm.github.io/niri/Configuration%3A-Key-Bindings.html#spawn-sh

**Comment by dead0user** [comment]
Thanks for help :-)
With full path it all works correct
